### PR TITLE
8311092: Please disable runtime/jni/nativeStack/TestNativeStack.java on armhf

### PR DESCRIPTION
**Pull request by vpa1977**
Changes:
- Add test exclusion for armhf, because os::current_frame() is not implemented

Testing:

```
AMD64:

==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java
                                                         1     1     0     0   
==============================
TEST SUCCESS

Stopping javac server
Finished building target 'test' in configuration 'linux-x86_64-server-release

ARMHF

Running test 'jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java'
Test results: no tests selected
Report written to /home/ubuntu/jdk/build/linux-arm-server-release/test-results/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java/html/report.html
Results written to /home/ubuntu/jdk/build/linux-arm-server-release/test-support/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java
Finished running test 'jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java'
Test report is stored in build/linux-arm-server-release/test-results/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java

==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java
                                                         0     0     0     0   
==============================
TEST SUCCESS


```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311092](https://bugs.openjdk.org/browse/JDK-8311092): Please disable runtime/jni/nativeStack/TestNativeStack.java on armhf (**Bug** - P3)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14750/head:pull/14750` \
`$ git checkout pull/14750`

Update a local copy of the PR: \
`$ git checkout pull/14750` \
`$ git pull https://git.openjdk.org/jdk.git pull/14750/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14750`

View PR using the GUI difftool: \
`$ git pr show -t 14750`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14750.diff">https://git.openjdk.org/jdk/pull/14750.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14750#issuecomment-1617097961)

### CODE REVIEW

File: test/hotspot/jtreg/ProblemList.txt
Comment (dholmes-ora) [inline test/hotspot/jtreg/ProblemList.txt:94]: When you ProblemList a test, the bugid you enter is for the bug that will fix the test (or product) and so allow you to remove it from the ProblemList again. If you want this test permanently disabled on ARM 32-bit then that should be done in the test via an `@requires` clause.

Thanks.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/ProblemList.txt
Comment (vpa1977) [reply test/hotspot/jtreg/ProblemList.txt]: Thank you!!! Updated PR to include requires condition rather than ProblemList entry

**Comment by vpa1977** [comment]
Testing (after requires changes)
```
ARM64:

$make test TEST="test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java" CONF=linux-x86_64-server-release
Using exact match for CONF=linux-x86_64-server-release (other matches are possible)
Building target 'test' in configuration 'linux-x86_64-server-release'
Test selection 'test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java', will run:
* jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java

Running test 'jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java'
Passed: runtime/jni/nativeStack/TestNativeStack.java
Test results: passed: 1
Report written to /home/vladimirp/git/jdk/build/linux-x86_64-server-release/test-results/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java/html/report.html
Results written to /home/vladimirp/git/jdk/build/linux-x86_64-server-release/test-support/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java
Finished running test 'jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java'
Test report is stored in build/linux-x86_64-server-release/test-results/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java

==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java
                                                         1     1     0     0   
==============================
TEST SUCCESS

ARMHF:

$ make test TEST="test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java" 
Building target 'test' in configuration 'linux-arm-server-release'
Creating jdk image
Creating CDS archive for jdk image for server
Test selection 'test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java', will run:
* jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java

Running test 'jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java'
Test results: no tests selected
Report written to /home/ubuntu/jdk/build/linux-arm-server-release/test-results/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java/html/report.html
Results written to /home/ubuntu/jdk/build/linux-arm-server-release/test-support/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java
Finished running test 'jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java'
Test report is stored in build/linux-arm-server-release/test-results/jtreg_test_hotspot_jtreg_runtime_jni_nativeStack_TestNativeStack_java

==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java
                                                         0     0     0     0   
==============================
TEST SUCCESS

Finished building target 'test' in configuration 'linux-arm-server-release'

```
Will double check with aarch64 (just in case)

File: test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java:2]: Please change the copyright to read "2022, 2023," - thanks.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Functional change is good, just need the copyright change adjusted.

Thanks

File: test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java
Comment (vpa1977) [reply test/hotspot/jtreg/runtime/jni/nativeStack/TestNativeStack.java]: Updated, thank you!!!

**Review by dholmes-ora — APPROVED** [review]
This looks good and trivial (so no need for a second review).

Thanks.

**Pull request closed without merging**
